### PR TITLE
fix(ruby): Correct handling of methods in Ruby modules

### LINKED ISSUE #13976: cloudrad: normalize method names in modules
currently module method names are treated as links (to themselves), correct this to match class/instance methods.

### PR DESCRIPTION
**Pull request by dazuma**
Fixes https://github.com/googleapis/google-cloud-ruby/issues/13976 in which methods in Ruby modules were mishandled, incorrectly treated as external links, and given the wrong (default) anchors rather than the `common.getHtmlId`-generated anchors.

Following a hint from @TheRoyalTnetennba I updated the item definitions to mark methods as `isEmbedded` when in a Ruby module. This PR also lifts the definition constants outside the `getDefinitions` function so they don't get recreated on every call. The results are visible in the updated Ruby goldens also present in this PR. No other languages should be affected.

### CODE REVIEW

File: testdata/ruby/obj/api/toc.yml [hunk 1]
```diff
@@ -5,6 +5,8 @@
     items:
     - name: Getting started
       href: index.md
+    - name: License
+      href: LICENSE.md
```
Comment (tbpg) [inline testdata/ruby/obj/api/toc.yml:9]: Unrelated to this PR, but it's surprising to me to include the license?

File: third_party/docfx/templates/devsite/UniversalReference.common.js [hunk 2]
```diff
@@ -344,37 +345,53 @@ function getDefinition(type) {
...
+  "property":     { inProperty: true,     typePropertyName: "inProperty",     id: "properties" },
+  "method":       { inMethod: true,       typePropertyName: "inMethod",       id: "methods" },
+  "event":        { inEvent: true,        typePropertyName: "inEvent",        id: "events" },
+  "operator":     { inOperator: true,     typePropertyName: "inOperator",     id: "operators" },
+  "eii":          { inEii: true,          typePropertyName: "inEii",          id: "eii" },
+  "member":       { inMember: true,       typePropertyName: "inMember",       id: "members"},
+  "function":     { inFunction: true,     typePropertyName: "inFunction",     id: "functions" },
+  "const":        { inConst: true,        typePropertyName: "inConst",        id: "consts", isEmbedded: true }
+};
+var rubyModuleItems = {
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/UniversalReference.common.js:378]: Will you add a comment saying this is the same as `classItems` except `methods` has `isEmbedded`? I missed that at first (and second) glance.

**Review by tbpg — APPROVED** [review]

File: testdata/ruby/obj/api/toc.yml (hunk 1 above)
Comment (dazuma) [reply testdata/ruby/obj/api/toc.yml:9]: It _is_ intentional. The license is included in the standard Ruby reference documentation for our libraries, and currently Ruby's cloud-rad strategy is to mirror the standard documentation layout. That said, if it's an issue, we can include logic to omit it.

File: third_party/docfx/templates/devsite/UniversalReference.common.js (hunk 2 above)
Comment (dazuma) [reply third_party/docfx/templates/devsite/UniversalReference.common.js:379]: done

**Pull request merged**